src/go/rpk/pkg/cli/cluster/health.go

```diff
@@ -21,89 +22,169 @@ import (
 	"github.com/twmb/types"
+	"go.uber.org/zap"
 )
 
+type healthResponse struct {
+	ClusterUUID               *string  `json:"cluster_uuid,omitempty" yaml:"cluster_uuid,omitempty"`
+	IsHealthy                 bool     `json:"is_healthy" yaml:"is_healthy"`
+	UnhealthyReasons          []string `json:"unhealthy_reasons" yaml:"unhealthy_reasons"`
+	ControllerID              int      `json:"controller_id" yaml:"controller_id"`
+	AllNodes                  []int    `json:"all_nodes" yaml:"all_nodes"`
+	NodesDown                 []int    `json:"nodes_down" yaml:"nodes_down"`
+	NodesInRecoveryMode       []int    `json:"nodes_in_recovery_mode" yaml:"nodes_in_recovery_mode"`
+	LeaderlessPartitions      []string `json:"leaderless_partitions" yaml:"leaderless_partitions"`
+	LeaderlessCount           *int     `json:"leaderless_count,omitempty" yaml:"leaderless_count,omitempty"`
+	UnderReplicatedCount      *int     `json:"under_replicated_count,omitempty" yaml:"under_replicated_count,omitempty"`
+	UnderReplicatedPartitions []string `json:"under_replicated_partitions" yaml:"under_replicated_partitions"`
+}
+
 func newHealthOverviewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	var watch, exit bool
 	cmd := &cobra.Command{
 		Use:   "health",
 		Short: "Queries cluster for health overview",
-		Long: `Queries health overview.
+		Long: `Queries cluster for health overview.
 
 Health overview is created based on the health reports collected periodically
 from all nodes in the cluster. A cluster is considered healthy when the
 following conditions are met:
 
-* all cluster nodes are responding
-* all partitions have leaders
-* the cluster controller is present
+  * All cluster nodes are responding
+  * All partitions have leaders
+  * The cluster controller is present
+
+If the cluster is reported as unhealthy, rpk will exit with code 10.
+`,
+		Example: `
+Basic usage, get cluster health information:
+  rpk cluster health
+
+Get cluster health information and watch for changes:
+  rpk cluster health --watch
+
+Get cluster health information and exit when the cluster is healthy:
+  rpk cluster health --exit-when-healthy
 `,
 		Args: cobra.ExactArgs(0),
 		Run: func(cmd *cobra.Command, _ []string) {
+			f := p.Formatter
+			if f.Kind != "text" && f.Kind != "help" && (watch || exit) {
+				out.Die("format type %q cannot be used along with --watch or --exit-when-healthy", f.Kind)
+			}
+			if h, ok := f.Help(healthResponse{}); ok {
+				out.Exit(h)
+			}
 			p, err := p.LoadVirtualProfile(fs)
 			out.MaybeDie(err, "rpk unable to load config: %v", err)
 			config.CheckExitCloudAdmin(p)
 
 			cl, err := adminapi.NewClient(cmd.Context(), fs, p)
 			out.MaybeDie(err, "unable to initialize admin client: %v", err)
 
+			resp, err := cl.ClusterUUID(cmd.Context())
+			var clusterUUID *string
+			if err != nil {
+				zap.L().Sugar().Warnf("unable to get cluster UUID for the cluster health report: %v; skipping collection and checking cluster health", err)
```

Comment:
Do we actually want to use the zap logger for the CLI?

Reply:
It's our current choice for all logging in rpk https://github.com/search?q=repo%3Aredpanda-data%2Fredpanda%20zap.L().&type=code 😅